catalog/gke/Kptfile

```diff
@@ -0,0 +1,84 @@
+apiVersion: kpt.dev/v1alpha1
+kind: Kptfile
+metadata:
+  name: gke
+packageMetadata:
+  shortDescription: GKE cluster with dedicated subnet
+openAPI:
+  definitions:
+    io.k8s.cli.setters.projects-namespace:
+      x-k8s-cli:
+        setter:
+          name: projects-namespace
+          value: projects
+    io.k8s.cli.setters.network-project-id:
+      x-k8s-cli:
+        setter:
+          name: network-project-id
+          value: network-project-id
+    io.k8s.cli.setters.networking-namespace:
+      x-k8s-cli:
+        setter:
+          name: networking-namespace
+          value: config-control
+    io.k8s.cli.setters.platform-project-id:
+      x-k8s-cli:
+        setter:
+          name: platform-project-id
+          value: platform-project-id
+    io.k8s.cli.setters.platform-namespace:
+      x-k8s-cli:
+        setter:
+          name: platform-namespace
+          value: config-control
+    io.k8s.cli.setters.cluster-name:
+      x-k8s-cli:
+        setter:
+          name: cluster-name
+          value: example-us-west4
+    io.k8s.cli.setters.environ:
```

Comment:
Is this just for users to label their env or does `gke.io/environ` do something more?

Reply:
It's a holdover from Tenant Factory, which supports multiple environments. I should probably rename it "env" since we've abandoned the "environ" term. All it does here is add a cluster label which makes it possible to get a filtered list by environment.

Reply:
Renamed to "environment" and added a comment in the cluster yaml.